.github/workflows/firmwares.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+jobs:
+  firmwares:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          path: trussed
+
+      - name: Checkout solokeys/solo2
+        run: git clone https://github.com/solokeys/solo2 solo2
+        #uses: actions/checkout@v3
+        #with:
+        #  repository: solokeys/solo2
+        #  path: solo2
+      
+      - name: Checkout Nitrokey/nitrokey-3-firmware
+        run: git clone https://github.com/Nitrokey/nitrokey-3-firmware nk3
+        #uses: actions/checkout@v3
+        #with:
+        #  repository: Nitrokey/nitrokey-3-firmware
+        #  path: nk3
+
+      - name: Install littlefs2-sys/micro-ecc-sys build dependencies
+        shell: bash
+        run: |
+          apt-get update && apt-get install sudo
+          env && pwd && sudo apt-get update -y -qq && sudo apt-get install -y -qq llvm libc6-dev-i386 libclang-dev clang git
+
+      - uses: fiam/arm-none-eabi-gcc@v1
+        with:
+          release: "9-2020-q2"
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: thumbv8m.main-none-eabi
+          override: true
+          components: llvm-tools-preview
+
+
+      - name: cargo install cargo-binutils
+        uses: actions-rs/install@v0.1
+        with:
+          crate: cargo-binutils
+          version: latest
+          use-tool-cache: true
+
+      - name: cargo install flip-link
+        uses: actions-rs/install@v0.1
+        with:
+          crate: flip-link
+          version: latest
+          use-tool-cache: true      
+
+      - name: Patch solo2 trussed-dependency
+        run: | 
+          echo "[patch.crates-io]" >> solo2/runners/lpc55/Cargo.toml
+          echo "trussed = { git = 'https://github.com/trussed-dev/trussed' }" >> solo2/runners/lpc55/Cargo.toml 
```

Comment:
yes, indeed and this could end up messy with commits/PRs whatsoever....
The 1st action actually does the checkout of the related repository (e.g., the PR's origin) into a local dir `trussed`, so why not use this one! done!